app/src/main/java/de/danoeh/antennapod/fragment/actions/FeedMultiSelectActionHandler.java

```diff
@@ -0,0 +1,139 @@
+package de.danoeh.antennapod.fragment.actions;
+
+import android.util.Log;
+
+import androidx.annotation.PluralsRes;
+import androidx.core.util.Consumer;
+
+import com.google.android.material.snackbar.Snackbar;
+
+import java.text.DecimalFormat;
+import java.text.DecimalFormatSymbols;
+import java.util.List;
+import java.util.Locale;
+
+import de.danoeh.antennapod.R;
+import de.danoeh.antennapod.activity.MainActivity;
+import de.danoeh.antennapod.core.storage.DBWriter;
+import de.danoeh.antennapod.dialog.RemoveFeedDialog;
+import de.danoeh.antennapod.fragment.preferences.dialog.PreferenceListDialog;
+import de.danoeh.antennapod.fragment.preferences.dialog.PreferenceSwitchDialog;
+import de.danoeh.antennapod.model.feed.Feed;
+import de.danoeh.antennapod.model.feed.FeedPreferences;
+
+public class FeedMultiSelectActionHandler {
+    private static final String TAG = "FeedSelectHandler";
+    private final MainActivity activity;
+    private final List<Feed> selectedItems;
+
+    public FeedMultiSelectActionHandler(MainActivity activity, List<Feed> selectedItems) {
+        this.activity = activity;
+        this.selectedItems = selectedItems;
+    }
+
+    public void handleAction(int id) {
+        if (id == R.id.remove_item) {
+            RemoveFeedDialog.show(activity, selectedItems, null);
+        } else if (id == R.id.keep_updated) {
+            keepUpdatedChecked();
+        } else if (id == R.id.autodownload) {
+            autoDownloadChecked();
+        } else if (id == R.id.autoDeleteDownload) {
+            autoDeleteEpisodesChecked();
+        } else if (id == R.id.playback_speed) {
+            playbackSpeedChecked();
+        } else {
+            Log.e(TAG, "Unrecognized speed dial action item. Do nothing. id=" + id);
+        }
+    }
+
+    private void autoDownloadChecked() {
```

Comment:
Checked? I would rather append the word Setting or Preference

Reply:
Okay, wan't my idea. Just kept the same naming from the original multi select screen. :)